### ISSUE TITLE
Calling show() multiple times results in an IllegalStateException

### ISSUE THREAD

**Issue by Diarrhio** [jama5262/CarouselView#20]
**Describe the bug**
Calling show() multiple times results in an IllegalStateException being thrown by SnapHelper::setupCallbacks, `throw new IllegalStateException("An instance of OnFlingListener already set.");`

**To Reproduce**
1. Load an initial data source, call show(). Works as expected
2. Load a subsequent data source and call show() again
3. See crash

**Expected behavior**
Should support calling show() multiple times to repopulate the carousel

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Android 11

**Smartphone (please complete the following information):**
 - Device: All

**Additional context**

Maybe I'm doing something wrong, but, I don't see a way to get the carousel to reload data other than calling show(). Here is my code:

```
        binding.carouselView.apply {
          size = assets.size
          resource = R.layout.view_asset_item
          autoPlay = false
          enableSnapping(true)
          hideIndicator(true)
          indicatorAnimationType = IndicatorAnimationType.NONE
          carouselOffset = OffsetType.CENTER
          setCarouselViewListener { view, position ->
            val assetItemView = view as AssetItemView
            assetItemView.setAsset(assets[position])
          }
          show()
        }
```

**Comment by Diarrhio** [comment]
Also tried this instead of calling show multiple times, but I was not able to get the data to reload:

```        binding.carouselView.size = assets.size
        requireView().findViewById<RecyclerView>(R.id.carouselRecyclerView).adapter!!.notifyDataSetChanged()
```

**Comment by rohanisme** [comment]
Same issues here as well 
can some one help me solve this

**Comment by Diarrhio** [comment]
@rohanisme I had to just fork the repo and make a bunch of changes to accommodate that and other needs. This library seems to be designed for static data that doesn't change

**Comment by rohanisme** [comment]
Can u share me the changed repo so i can use the same

**Comment by Diarrhio** [comment]
@rohanisme I can't unfortunately, but it wasn't hard

**Comment by ashharjamil** [comment]
before apply onFlingListener null

binding.carouselView.findViewById<RecyclerView>(R.id.carouselRecyclerView)..onFlingListener = null

**Comment by derysudrajat** [comment]
I've got the same problem, I just made it static, I am using live data to load the update news data when it was added to the bookmark and really not affected in the carousel, so I just create an instance of my data, and check it, if it was empty add and load data carousel
```kt
private var carouselData = listOf<ArticlesItem?>()
```
in observer
```kt
model.getNews().observe(viewLifecycleOwner, {
     if (carouselData.isEmpty()) {
         carouselData = it.articles!!
         setDataCarousel(carouselData)
     }    
})
```

I've tried to remove all data but it doesn't work

**Comment by krishh95** [comment]
> before apply set onFlingListener null
> 
> binding.carouselView.findViewById(R.id.carouselRecyclerView).onFlingListener = null

this was better  solution to resolve issue

**Comment by its-the-tom** [comment]
val v: RecyclerView = binding.videoSlider!!.findViewById(R.id.carouselRecyclerView)
v.onFlingListener = null


it's is working perfectly

**Comment by jama5262** [comment]
Solved in #26
